src/services/TeamAssignmentService.ts

```diff
@@ -0,0 +1,37 @@
+import { NewActiveMatchInput } from "../repositories/ActiveMatchRepository/types";
+import { PlayerInQueue } from "../repositories/QueueRepository/types";
+import { Team } from "../types/common";
+
+export async function createRandomTeams(ballchasers: PlayerInQueue[]): Promise<Array<NewActiveMatchInput>> {
+  const sortedBallChaser = ballchasers.sort((o, b) => (o.mmr > b.mmr ? -1 : 1));
+
+  const blueTeam: number[] = [];
+  const orangeTeam: number[] = [];
+  const activeMatch: NewActiveMatchInput[] = [];
+
+  //This splits the teams with the minimal difference in the sum of MMR.
+  //This actually splits the ball chasers into two arrays that aren't
+  //neccessarily even arrays.  So the If statements that say .lenght /2
+  //is ensuring that there aren't 4 players on one team and two on the other.
+  sortedBallChaser.forEach((p) => {
+    if (blueTeam.reduce((sum, current) => sum + current, 0) > orangeTeam.reduce((sum, current) => sum + current, 0)) {
```

Comment:
Instead of running two reduces each loop, can we just keep track of the team's mmr as like a counter instead? In other words have a `let orangeTeamMmr: number = 0` and `let blueTeamMmr: number = 0` and just increment the number as a player joins?

Reply:
Fixed, I also had to add a teamcounter for each team to ensure we don't go over the team boundaries.